packages/react-components/src/graph-visualizer/graph.ts

```diff
@@ -0,0 +1,523 @@
+/* eslint-disable prefer-const */
```

Comment:
This file was large enough github didn't want to load it for the diff review so it's probably worth seeing if we can break it up.

Reply:
Done,

Reply:
I am always in favor of code decomposition.